lib/livebook_web/live/app_helpers.ex

```diff
@@ -79,4 +83,353 @@ defmodule LivebookWeb.AppHelpers do
       confirm_icon: "delete-bin-6-line"
     )
   end
+
+  @doc """
+  Builds a changeset for app Dockerfile configuration.
+  """
+  @spec docker_config_changeset(map()) :: Ecto.Changeset.t()
+  def docker_config_changeset(attrs \\ %{}) do
+    default_image = Livebook.Config.docker_images() |> hd()
+
+    data = %{deploy_all: false, docker_tag: default_image.tag, zta_provider: nil, zta_key: nil}
+
+    zta_types =
+      for provider <- Livebook.Config.identity_providers(),
+          not provider.read_only,
+          do: provider.type
+
+    types = %{
+      deploy_all: :boolean,
+      docker_tag: :string,
+      zta_provider: Ecto.ParameterizedType.init(Ecto.Enum, values: zta_types),
+      zta_key: :string
+    }
+
+    cast({data, types}, attrs, [:deploy_all, :docker_tag, :zta_provider, :zta_key])
+    |> validate_required([:deploy_all, :docker_tag])
+  end
```

Comment:
Let's move this function and the Dockerfile generator to Livebook.Hub.Dockerfile or something? There is nothing view specific on them. :)

Reply:
Sounds good :)